core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -103,6 +104,81 @@ private[spark] class AppStatusListener(
     }
   }
 
+  // visible for tests
+  private[spark] def recoverLiveEntities(): Unit = {
+    if (!live) {
+      kvstore.view(classOf[JobDataWrapper])
+        .asScala.filter(_.info.status == JobExecutionStatus.RUNNING)
+        .map(_.toLiveJob).foreach(job => liveJobs.put(job.jobId, job))
+
+      kvstore.view(classOf[StageDataWrapper]).asScala
+        .filter { stageData =>
+          stageData.info.status == v1.StageStatus.PENDING ||
```

Comment:
Logically, `activeTasks > 0` is counted in `liveStages` regardless of status:

https://github.com/apache/spark/blob/5a512e86e94593bc004a35101ad6497e20c13e0a/core/src/main/scala/org/apache/spark/status/AppStatusListener.scala#L605-L614

https://github.com/apache/spark/blob/5a512e86e94593bc004a35101ad6497e20c13e0a/core/src/main/scala/org/apache/spark/status/AppStatusListener.scala#L743-L748

... except status is SKIPPED:

https://github.com/apache/spark/blob/5a512e86e94593bc004a35101ad6497e20c13e0a/core/src/main/scala/org/apache/spark/status/AppStatusListener.scala#L408-L429

so the condition is actually much complicated than that.

Reply:
Good catch here ! And I attached another possible condition for this:

`(stageData.info.numActiveTasks > 0 && stageData.info.status != v1.StageStatus.SKIPPED)`